addons/survey/static/src/js/survey_form.js

```diff
@@ -1167,10 +1171,20 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
     // -------------------------------------------------------------------------
 
     _showErrors: function (errors) {
-        var self = this;
-        var errorKeys = _.keys(errors);
+        const self = this;
+        const errorKeys = _.keys(errors);
         _.each(errorKeys, function (key) {
-            self.$("#" + key + '>.o_survey_question_error').append($('<p>', {text: errors[key]})).addClass("slide_in");
+            let msg = errors[key];
+            // Matrix missing rows highlight
+            if (errors[key] instanceof Object){
```

Comment:
**To discuss**: The signature of the `_showErrors` function is not really clear but I think it expects a json object mapping a number to a string. I would maybe avoid mixing different data types and put the code highlighting the invalid rows inside the switch case statement (in an auxiliary function ?) as it's something specific to the matrices. The `_showErrors` function will then only have the responsibility to show the generic error message.